rmw/include/rmw/types.h

```diff
@@ -192,14 +226,30 @@ enum RMW_PUBLIC_TYPE rmw_qos_durability_policy_t
   RMW_QOS_POLICY_DURABILITY_VOLATILE
 };
 
+enum RMW_PUBLIC_TYPE rmw_qos_liveliness_policy_kind_t
+{
+  RMW_QOS_POLICY_LIVELINESS_SYSTEM_DEFAULT,
+  RMW_QOS_POLICY_LIVELINESS_AUTOMATIC,
+  // TODO(eknapp) - re-enable these once these types of liveliness are supported
+  // RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_PARTICIPANT,
+  // RMW_QOS_POLICY_LIVELINESS_MANUAL_BY_TOPIC
+};
+
 /// ROS MiddleWare quality of service profile.
 typedef struct RMW_PUBLIC_TYPE rmw_qos_profile_t
 {
   enum rmw_qos_history_policy_t history;
   /// Size of the message queue.
   size_t depth;
+
   enum rmw_qos_reliability_policy_t reliability;
   enum rmw_qos_durability_policy_t durability;
+  struct rmw_time_t deadline;
+  struct rmw_time_t lifespan;
+
+  enum rmw_qos_liveliness_policy_kind_t liveliness;
+  struct rmw_time_t liveliness_lease_duration;
```

Comment:
Could we add this to the rmw_qos_liveliness_policy_t since they are both required and related?

Reply:
See above comment about the existing pattern of splitting HISTORY policy - if we want to group it's fine with me, but just want you to be aware of that while evaluating this

Reply:
Okay, let's be consistent with `rmw_qos_history_policy_t` + `size_t`, and _not_ group `rmw_qos_liveliness_policy_kind_t` + `rmw_time_t` into `rmw_qos_liveliness_policy_t`.